crates/pathfinder/src/state/sync.rs

```diff
@@ -44,7 +44,46 @@ impl Default for State {
     }
 }
 
+pub struct PendingData {
+    inner: RwLock<Option<(Box<PendingBlock>, Box<sequencer::reply::StateUpdate>)>>,
```

Comment:
This makes me worried that we require both of these to be ready, even though we have no such guarantee from the api.

Also, I don't understand why there are `Box<_>` around these? If they are large, there's no point in cloning them (should be ref counted). If they are not large, should just be inline and copied?

Since there's no way to look at these and see if they are still valid, is the procedure that they will be cleared once the block matures to an L2_ACCEPTED?

Reply:
I think I just `Box<_>` because the enum they came from was `Box` and I didn't want to deref and clone again. I'll remove the box.

They do indeed get cleared every time there is a block change -- either a reorg or a new block arrives. It would really be confusing if our pending queries return data that is older than latest for example so this has to be the case.

Reply:
If they're not both ready, then neither gets populated. We can consider them independently I guess, but that makes things quite complex I think.

Reply:
I think a `Arc` should be considered since there are the transactions and receipts on block, statediff is a lighter.

Reply:
If you're thinking about an inner Arc then it would defeat the idea of the lock having exclusive ownership. If you're thinking an outer Arc, then there is one already.

Reply:
I meant an inner arc. I don't understand the argument about lock having exclusive ownership. I need to ship the stateupdate over a channel to be executed on one python process.

Reply:
Inner arc received. I didn't realize I had already commented about that validity period so resolving this in favor of: https://github.com/eqlabs/pathfinder/pull/446#discussion_r923523358